File: tests/fixtures/projects/gotest-basic-coverage/classifier.go
// Package gotestbasiccoverage is the SuT for the gotest-basic-coverage
// fixture. Two source files (`classifier.go`, `arithmetic.go`) give
// `cover.out` interesting block structure: multiple files, multiple
// functions, multiple branches, with one branch deliberately
// uncovered so the report carries a concrete uncovered region.
package gotestbasiccoverage

// Classify returns "positive", "zero", or "negative" depending on the
// sign of v. The negative branch is intentionally not exercised by the
// test suite — fixture contract — so `cover.out` carries an uncovered
// region for it.
func Classify(v int) string {
	if v > 0 {
		return "positive"
	}
	if v == 0 {
		return "zero"
	}
	return "negative"
}

File: tests/fixtures/projects/cargo-test-basic-coverage/src/classifier.rs
//! Three-branch classifier. The negative branch is intentionally not
//! exercised by any test so `coverage.lcov` carries at least one
//! uncovered region.

pub fn classify(v: i32) -> &'static str {
    if v > 0 {
        "positive"
    } else if v == 0 {
        "zero"
    } else {
        // Fixture contract: this branch is intentionally not covered by
        // any test. Do NOT add a test that hits a negative value.
        "negative"
    }
}
